src/lib/spinel/radio_spinel_impl.hpp

```diff
@@ -1946,22 +1956,43 @@ otError RadioSpinel<InterfaceType, ProcessContextType>::Transmit(otRadioFrame &a
+                                                    SPINEL_DATATYPE_BOOL_S            // SkipAes
+                                                        SPINEL_DATATYPE_UINT32_S      // TxDelay
+                                                            SPINEL_DATATYPE_UINT32_S, // TxDelayBaseTime
+                        mTransmitFrame->mPsdu, mTransmitFrame->mLength, mTransmitFrame->mChannel,
+                        mTransmitFrame->mInfo.mTxInfo.mMaxCsmaBackoffs, mTransmitFrame->mInfo.mTxInfo.mMaxFrameRetries,
+                        mTransmitFrame->mInfo.mTxInfo.mCsmaCaEnabled, mTransmitFrame->mInfo.mTxInfo.mIsHeaderUpdated,
+                        mTransmitFrame->mInfo.mTxInfo.mIsARetx, mTransmitFrame->mInfo.mTxInfo.mIsSecurityProcessed,
+                        mTransmitFrame->mInfo.mTxInfo.mTxDelay, mTransmitFrame->mInfo.mTxInfo.mTxDelayBaseTime);
+    }
+    else
+    {
+        error = Request(SPINEL_CMD_PROP_VALUE_SET, SPINEL_PROP_STREAM_RAW,
+                        SPINEL_DATATYPE_DATA_WLEN_S                               // Frame data
+                            SPINEL_DATATYPE_UINT8_S                               // Channel
+                                SPINEL_DATATYPE_UINT8_S                           // MaxCsmaBackoffs
+                                    SPINEL_DATATYPE_UINT8_S                       // MaxFrameRetries
+                                        SPINEL_DATATYPE_BOOL_S                    // CsmaCaEnabled
                                             SPINEL_DATATYPE_BOOL_S                // IsARetx
                                                 SPINEL_DATATYPE_BOOL_S            // SkipAes
                                                     SPINEL_DATATYPE_UINT32_S      // TxDelay
                                                         SPINEL_DATATYPE_UINT32_S, // TxDelayBaseTime
-                    mTransmitFrame->mPsdu, mTransmitFrame->mLength, mTransmitFrame->mChannel,
-                    mTransmitFrame->mInfo.mTxInfo.mMaxCsmaBackoffs, mTransmitFrame->mInfo.mTxInfo.mMaxFrameRetries,
-                    mTransmitFrame->mInfo.mTxInfo.mCsmaCaEnabled, mTransmitFrame->mInfo.mTxInfo.mIsHeaderUpdated,
-                    mTransmitFrame->mInfo.mTxInfo.mIsARetx, mTransmitFrame->mInfo.mTxInfo.mIsSecurityProcessed,
-                    mTransmitFrame->mInfo.mTxInfo.mTxDelay, mTransmitFrame->mInfo.mTxInfo.mTxDelayBaseTime);
+                        mTransmitFrame->mPsdu, mTransmitFrame->mLength, mTransmitFrame->mChannel,
+                        mTransmitFrame->mInfo.mTxInfo.mMaxCsmaBackoffs, mTransmitFrame->mInfo.mTxInfo.mMaxFrameRetries,
+                        mTransmitFrame->mInfo.mTxInfo.mCsmaCaEnabled, mTransmitFrame->mInfo.mTxInfo.mIsARetx,
+                        mTransmitFrame->mInfo.mTxInfo.mIsSecurityProcessed, mTransmitFrame->mInfo.mTxInfo.mTxDelay,
+                        mTransmitFrame->mInfo.mTxInfo.mTxDelayBaseTime);
+    }
```

Comment:
Thank @dismirlian 

I am not sure this can work. 

- The main change in https://github.com/openthread/openthread/pull/6849 is that it adds a new behavior and new flag `mIsHeaderUpdated ` in the interaction between OT core (mac/sub-mac) and the radio platform 
  - This flag may be set/cleared by either side.
- This in turn required changes to the spinel frame formats between host and RCP to add support for the new flag. 
- The changes you have in this PR make the spinel frame to be parsed correctly based on the RCP API version, but we still need to convey the flag between radio platform and OT stack.
  - I don't think there is any easy default value or behavior we can use for this flag (when RCP version is old and does not support it).
  - I order for such a backward compatibility situation to work, we need OT stack itself be aware of older versions of radio platform APIs and behave differently based on that. This can become rather complex and not a desirable model.

Reply:
Hi @abtink thanks for reviewing.

I think this works because:
- Here, `mIsHeaderUpdated` is taken from the buffered TX frame. This may work because the the old RCP never touches this flag (is unaware of it), so `HandleTransmitDone` takes care of setting the returned flag exactly as the host transmitted it. This relies on the fact that the transmissions are done one by one.
https://github.com/openthread/openthread/blob/10bca528a581fdf3a15cd1414baf43194cd45652/src/lib/spinel/radio_spinel_impl.hpp#L1909-L1912
- Here, the initialization code checks that either the required radio caps don't include `OT_RADIO_CAPS_TRANSMIT_SEC`, or the RCP API version is >= 4 (which handles the `mIsHeaderUpdated` flag correctly):
https://github.com/openthread/openthread/blob/10bca528a581fdf3a15cd1414baf43194cd45652/src/lib/spinel/radio_spinel_impl.hpp#L395-L396

Of course, I may be missing something. What do you think?

Reply:
Btw, this changes are working with a RCP with API version = 1, and the latest commits. I haven't done much stress testing yet, so I may well be wrong.

Reply:
Thanks @dismirlian. 

I see what you mean. You are adding the backward compatibility for `API_VERSION=1` only when capability `CAPS_TRANSMIT_SEC` is not used.  Or basically it is only limited when OT is built as **Thread 1.1** and not when built as **Thread 1.2** (or 1.3...) :)

I agree under Thread 1.1 this may work (with current code), but in the general case we still require `API_VERION=4` (I don't see any way to make it bckward compatible under general case when `CAPS_TRANSMIT_SEC` is used).
 - One immediate thought on this: I feel we should keep the `SPINEL_MIN_HOST_SUPPORTED_RCP_API_VERSION` as `4` in `spine.h` since we still require 4 as minimum for the general case.

Reply:
Some more thoughts on this:
- I know adding code to handle backward compatibility (under certain  conditions) may help let host side firmware to be updated to newer version without updating the RCP firmware. 
  - But the main challenge I see with this is that it it makes the code more complex and require us to maintain and test it 
  - I feel it can get even more complex if/when other incompatible changes are added in future.
- On the other hand, it may be better to require an update of RCP when we have incompatible change. Of course, we should aim to avoid/minimize incompatible changes.  

My first preference/suggestion would be to not add such a code (require to go through the pain of updating of RCP) 
Second preference is to add it under an specific OT config so that it can enabled and used during the transition of firmwares but then can be disabled and maybe deprecated/removed in longer term.

Thoughts?